.ruff.toml

```diff
@@ -0,0 +1,130 @@
+    "D401", # first line should be in imperative mood -> TBD
+    "D205", # 1 blank line required between summary line and description -> not a problem in docs
+    "D100", # missing docstring in public module -> TBD
+    "D101", # missing docstring in public class -> TBD
+    "D102", # missing docstring in public method -> TBD
+    "D103", # missing docstring in public function -> TBD
+    "D104", # missing docstring in public package -> TBD
+    "D105", # missing docstring in magic method -> TBD
+    "D107", # missing docstring in __init__ -> TBD
+    "D417", # missing argument descriptions in the docstring -> TBD
+    # UP
+    "UP012", # unnecessary call to 'encode' as UTF-8 -> TBD
+    # ANN 
+    "ANN", # do we want to have type annotations? -> TBD
+    "S", # mainly unncessary rule -> TBD
+    "FBT", 
+    "B", # need to be fixed by hand -> TBD
+    "C4", # flake8 plugin -> need to be fixed by hand -> TBD
+    "DTZ005", # datetime.datetime.now() called without a tz argument
+    "INP", # will be fixed when test are moved
+    "T20", # will be fixed when logging is implemented
+    "PT", # will be solved when test infrastructure is refactored
+    "RET", # need to be handled manually -> TBD (separate issue)
+    "SLF", # need to be handled manually -> TBD (separate issue: #225 )
+    "SIM", # simplifications need to be done manually -> TBD 
+    "ARG", # need to be handled manually -> #226
+    "TD", # todos should be formulated as issues -> do we allow todos in the code? -> #227
+    "FIX", # same as with TD
+    "ERA", # plenty of outcommented code -> TBD
+]
+
+# Allow fix for all enabled rules (when `--fix`) is provided.
+fixable = ["ALL"]
+unfixable = []
+
+[lint.pydocstyle]
+convention = "google"  # Accepts: "google", "numpy", or "pep257".
+
+[format]
+quote-style = "single"
+indent-style = "space"
+skip-magic-trailing-comma = false
+
+# Enable auto-formatting of code examples in docstrings. Markdown,
+# reStructuredText code/literal blocks and doctests are all supported.
+#
+# This is currently disabled by default, but it is planned for this
+# to be opt-out in the future.
+docstring-code-format = true
+
+# Set the line length limit used when formatting code snippets in
+# docstrings.
+#
+# This only has an effect when the `docstring-code-format` setting is
+# enabled.
+docstring-code-line-length = "dynamic"
+
+
+[lint.per-file-ignores]
+"acoular/__init__.py" = ["F401"] # unused import -> TBD
+"acoular/demo/__init__.py" = ["F401"] # unused import -> TBD
+"acoular/configuration.py" = ["F401"] # unused import (will be solved by issue #212)
+"acoular/nidaqimport.py" = ["F841"] # unused variable (really unused?) -> TBD
+"acoular/tbeamform.py" = ["F841"] # unused variable (really unused?) -> TBD
+"acoular/fbeamform.py" = ["W605"] # invalid escape sequence -> TBD
+"acoular/tests/test_tprocess.py" = ["F403"] # star import 
+"acoular/tests/*" = ["D"] # exclude docstring linting
+"acoular/version.py" = ["D"] # exclude docstring linting
```

Comment:
Do all of these concern multiple errors or only some instances in these files? If the latter is the case, it would be probably better to use `# noqa: ["X000"]` directives.